esphome/components/touchscreen/__init__.py

```diff
@@ -3,44 +3,84 @@
 
 from esphome.components import display
 from esphome import automation
-from esphome.const import CONF_ON_TOUCH
+from esphome.const import CONF_ON_TOUCH, CONF_ON_RELEASE, CONF_DIMENSIONS, CONF_ROTATION
 from esphome.core import coroutine_with_priority
 
-CODEOWNERS = ["@jesserockz"]
+CODEOWNERS = ["@jesserockz", "@nielsnl68"]
 DEPENDENCIES = ["display"]
 
 IS_PLATFORM_COMPONENT = True
 
 touchscreen_ns = cg.esphome_ns.namespace("touchscreen")
 
-Touchscreen = touchscreen_ns.class_("Touchscreen")
+Touchscreen = touchscreen_ns.class_("Touchscreen", cg.PollingComponent)
 TouchRotation = touchscreen_ns.enum("TouchRotation")
 TouchPoint = touchscreen_ns.struct("TouchPoint")
+TouchPoints_t = cg.std_vector.template(TouchPoint)
+TouchPoints_t_const_ref = TouchPoints_t.operator("ref").operator("const")
 TouchListener = touchscreen_ns.class_("TouchListener")
 
 CONF_DISPLAY = "display"
 CONF_TOUCHSCREEN_ID = "touchscreen_id"
+CONF_REPORT_INTERVAL = "report_interval"  # not used yet:
+CONF_SWAP_X_Y = "swap_x_y"
+CONF_ON_UPDATE = "on_update"
+
+DISPLAY_ROTATIONS = {
+    0: touchscreen_ns.ROTATE_0_DEGREES,
+    90: touchscreen_ns.ROTATE_90_DEGREES,
+    180: touchscreen_ns.ROTATE_180_DEGREES,
+    270: touchscreen_ns.ROTATE_2700_DEGREES,
+}
+
+
+def validate_rotation(value):
+    value = cv.string(value)
+    if value.endswith("°"):
+        value = value[:-1]
+    return cv.enum(DISPLAY_ROTATIONS, int=True)(value)
 
 
 TOUCHSCREEN_SCHEMA = cv.Schema(
     {
-        cv.GenerateID(CONF_DISPLAY): cv.use_id(display.DisplayBuffer),
+        cv.Required(CONF_DISPLAY): cv.use_id(display.DisplayBuffer),
```

Comment:
Lets not break the config here until we are making changes for it to become optional
```suggestion
        cv.GenerateID(CONF_DISPLAY): cv.use_id(display.DisplayBuffer),
```